scripts/deploy_006_migration.py

```diff
@@ -0,0 +1,185 @@
+      ogv_epoch, 
+      ogv_min_staking, 
+      REWARDS_PROXY_ADDRESS, 
+      migrator_proxy.address, 
+      FROM_DEPLOYER
+    )
+    print("OGVStaking Implementation deployed at {}".format(veogv_impl.address))
+
+    # Deploy xOGN implementation
+    ogn_min_staking = 7 * 24 * 60 * 60 # 7 days
+    ogn_epoch = 24 * 60 * 60 # 1 day
+    xogn_impl = ExponentialStaking.deploy(
+      OGN_ADDRESS, 
+      ogn_epoch, 
+      ogn_min_staking, 
+      ogn_rewardssource_proxy.address,
+      FROM_DEPLOYER
+    )
+    print("ExponentialStaking implementation deployed at {}".format(xogn_impl.address))
+    xogn_proxy.initialize(
+      xogn_impl.address,
+      TIMELOCK,
+      b'',
+      FROM_DEPLOYER
+    )
+    print("ExponentialStaking proxy initialized")
+
+    # Deploy FixedRateRewardsSource
+    rewards_per_second = 100 # TODO: Decide on the params
+    ogn_rewardssource_impl = FixedRateRewardsSource.deploy(OGN_ADDRESS, FROM_DEPLOYER)
+    print("FixedRateRewardsSource implementation deployed at {}".format(ogn_rewardssource_impl.address))
+    # ogn_rewardssource = FixedRateRewardsSource.at(ogn_rewardssource_proxy.address)
+    impl_init_data = ogn_rewardssource_impl.initialize.encode_input(STRATEGIST, xogn_proxy.address, rewards_per_second)
+    ogn_rewardssource_proxy.initialize(
+      ogn_rewardssource_impl.address,
+      TIMELOCK,
+      impl_init_data,
+      FROM_DEPLOYER
+    )
+    print("FixedRateRewardsSource proxy initialized")
+
+    # Deploy Migrator implementation
+    migrator_impl = Migrator.deploy(
+      OGV_PROXY_ADDRESS,
+      OGN_ADDRESS,
+      VEOGV_PROXY_ADDRESS,
+      xogn_proxy.address,
+      FROM_DEPLOYER
+    )
+    print("Migrator implementation deployed at {}".format(migrator_impl.address))
+    migrator_proxy.initialize(
+      migrator_impl.address,
+      TIMELOCK,
+      b'',
+      FROM_DEPLOYER
+    )
+    print("Migrator proxy initialized")
+    # migrator = Migrator.at(migrator_proxy.address)
+
+    if is_mainnet:
+      # Verify contract on etherscan on mainnet
+      print("Verifying contract on Etherscan...")
+
+      # Verify proxies
+      FixedRateRewardsSourceProxy.publish_source(ogn_rewardssource_proxy)
+      MigratorProxy.publish_source(migrator_proxy)
+      ExponentialStakingProxy.publish_source(xogn_proxy)
+
+      # Verify Governance
+      Governance.publish_source(xogn_governance)
+
+      # Verify implementations
+      OgvStaking.publish_source(veogv_impl)
+      FixedRateRewardsSource.publish_source(ogn_rewardssource_impl)
+      Migrator.publish_source(migrator_impl)
+      ExponentialStaking.publish_source(xogn_impl)
+      
+    return {
+      'name': 'Deploy OGV-OGN Migration contracts and OGN Governance',
+      'actions': [
+        ### OGV Governance
+        {
+          # Upgrade veOGV
+          'contract': veogv_proxy,
+          'signature': 'upgradeTo(address)',
+          'args': [veogv_impl.address]
+        },
+
+        ### OGN Governance
```

Comment:
These OGN governance grants need to be in a separate deploy file that we will run later.

However, we should remove governance from OGV in this deploy file.

Reply:
Noted, will include it in the foundry script

Reply:
Scraped the foundry script since I encountered some random issues locally. Have split the xOGN from this and added another brownie script (`deploy_007_xogn_governance`)